### PR TITLE
Add more tensor backward operations

### PR DESCRIPTION
**Pull request by asuhan**

### CODE REVIEW

File: torch_xla/csrc/convolution.cpp [hunk 1]
```diff
@@ -224,7 +218,7 @@ xla::XlaOp BuildThnnConv2dBackwardWeight(
   const auto padding_config =
       XlaHelpers::MakeXlaPaddingConfig(XlaHelpers::I64List(padding_attr));
 
-  auto builder = grad.builder();
+  auto builder = grad_output.builder();
```
Comment (dlibenzi) [inline torch_xla/csrc/convolution.cpp]: We are a bit all over with auto and non-auto. I kinda like types, unless they take two screens to show/type.
Maybe we can have a rule that we use auto for templated types and for for-iteration-variables.

File: torch_xla/csrc/tensor.cpp [hunk 2]
```diff
@@ -603,6 +607,66 @@ XLATensor XLATensor::avg_pool2d(
       GetDevice());
 }
 
+XLATensor XLATensor::mm(const XLATensor& input, const XLATensor& weight,
+                        bool use_full_conv_precision) {
+  return Create(
+      ir::NodeOperand(ir::ops::MatMulOp(ir::NodeOperand(input.GetIrNode()),
```
Comment (dlibenzi) [inline torch_xla/csrc/tensor.cpp]: You can drop the NodeOperand() ctor. Now is implicit.

File: torch_xla/csrc/tensor.cpp [hunk 3]
```diff
@@ -603,6 +607,66 @@ XLATensor XLATensor::avg_pool2d(
...
+}
+
+XLATensor XLATensor::avg_pool2d_backward(
+    const XLATensor& out_backprop, const XLATensor& input,
+    tensorflow::gtl::ArraySlice<const xla::int64> kernel_size,
+    tensorflow::gtl::ArraySlice<const xla::int64> stride,
+    tensorflow::gtl::ArraySlice<const xla::int64> padding,
+    bool count_include_pad) {
+  return Create(ir::NodeOperand(std::make_shared<ir::ops::AvgPool2dBackward>(
+                    ir::NodeOperand(out_backprop.GetIrNode()),
```
Comment (dlibenzi) [inline torch_xla/csrc/tensor.cpp]: Same for NodeOperand.

Comment (dlibenzi) [reply torch_xla/csrc/tensor.cpp]: Check everywhere in this PR.

File: torch_xla/csrc/convolution.cpp (hunk 1 above)
Comment (asuhan) [reply torch_xla/csrc/convolution.cpp]: We'll do this later.

File: torch_xla/csrc/tensor.cpp (hunk 3 above)
Comment (asuhan) [reply torch_xla/csrc/tensor.cpp]: Done.

**Pull request merged**